internal/cmd/login.go

```diff
@@ -24,57 +23,42 @@ import (
 	"github.com/infrahq/infra/internal/certs"
 	"github.com/infrahq/infra/internal/cmd/types"
 	"github.com/infrahq/infra/internal/format"
-	"github.com/infrahq/infra/internal/generate"
 	"github.com/infrahq/infra/internal/logging"
 )
 
 type loginCmdOptions struct {
 	Server             string
-	AccessKey          string
-	Provider           string
 	SkipTLSVerify      bool
 	TrustedCertificate string
 	TrustedFingerprint string
 	NonInteractive     bool
 	NoAgent            bool
+	User               string
 }
 
-type loginMethod int8
-
-const (
-	localLogin loginMethod = iota
-	deviceLogin
-	oidcLogin
-)
-
-const cliLoginRedirectURL = "http://localhost:8301"
+const DeviceFlowMinVersion = "0.18.0"
 
 func newLoginCmd(cli *CLI) *cobra.Command {
 	var options loginCmdOptions
 
 	cmd := &cobra.Command{
-		Use:   "login [SERVER]",
-		Short: "Login to Infra",
-		Long:  "Login to Infra and start a background agent to keep local configuration up-to-date",
-		Example: `# By default, login will prompt for all required information.
-$ infra login
-
-# Login to a specific server
-$ infra login infraexampleserver.com
-
-# Login with a specific identity provider
-$ infra login --provider okta
-
-# Login with an access key
-$ export INFRA_ACCESS_KEY=1M4CWy9wF5.fAKeKEy5sMLH9ZZzAur0ZIjy
-$ infra login
-
-# Login with pre-set provider and server
-$ export INFRA_SERVER=example.infrahq.com
-$ export INFRA_PROVIDER=google
-$ infra login`,
+		Use:     "login [SERVER]",
+		Short:   "Login to Infra",
 		Args:    MaxArgs(1),
 		GroupID: groupCore,
+		Example: `
+# Login
+infra login example.infrahq.com
+
+# Login with username (will prompt for password)
+infra login example.infrahq.com --user foo@gmail.com
+
+# Login with username and password in (non-interactive)
+export INFRA_SERVER=example.infrahq.com
+export INFRA_USER=foo@gmail.com
+export INFRA_PASSWORD=p4ssw0rd
```

Comment:
There's no longer any `INFRA_PASSWORD` env var, right? Seems like that was removed along with the flag. Do we still want the password to be set exclusively from env var?

Reply:
Oops, it still exists, but was in a different place than I expected.